src/services/Provider.ts

```diff
@@ -423,21 +423,29 @@ export class Provider {
     assets: ComputeAsset[],
     algorithm: ComputeAlgorithm,
     computeEnv: string,
-    validUntil: number,
     providerUri: string,
     accountId: string,
+    chainId: number,
+    token: string,
+    maxJobDuration: number,
     signal?: AbortSignal
   ): Promise<ProviderComputeInitializeResults> {
     const providerEndpoints = await this.getEndpoints(providerUri)
     const serviceEndpoints = await this.getServiceEndpoints(
       providerUri,
       providerEndpoints
     )
+    const payment = {
```

Comment:
it would be nice to add some types for payment and provider data, like we have in the node for eg:

https://github.com/oceanprotocol/ocean-node/blob/main/src/%40types/C2D/C2D.ts#L167